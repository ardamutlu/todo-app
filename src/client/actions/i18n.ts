import { State } from '../reducers/i18n';

export type Actions = ReturnType<typeof setLanguage>;

export const SET_LANGUAGE = 'SET_LANGUAGE';

export const setLanguage = (lang: State['lang']) =>
  ({
    type: SET_LANGUAGE,
    payload: {
      lang
    }
  } as const);
