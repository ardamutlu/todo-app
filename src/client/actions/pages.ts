import { State } from '../reducers/pages';

export type Actions =
  | ReturnType<typeof setEnv>
  | ReturnType<typeof setIsMobile>
  | ReturnType<typeof setBaseUrl>
  | ReturnType<typeof resetPageStatus>
  | ReturnType<typeof loadAppProcess>
  | ReturnType<typeof loadAppProcessSuccess>
  | ReturnType<typeof loadAppProcessFailure>
  | ReturnType<typeof loadHomePage>
  | ReturnType<typeof loadHomePageSuccess>
  | ReturnType<typeof loadHomePageFailure>
  | ReturnType<typeof loadDocumentationPage>
  | ReturnType<typeof loadDocumentationPageSuccess>
  | ReturnType<typeof loadDocumentationPageFailure>;

export const SET_ENV = 'SET_ENV';
export const setEnv = (env: State['env']) =>
  ({
    type: SET_ENV,
    payload: {
      env
    }
  } as const);

export const SET_IS_MOBILE = 'SET_IS_MOBILE';
export const setIsMobile = (isMobile: State['isMobile']) =>
  ({
    type: SET_IS_MOBILE,
    payload: {
      isMobile
    }
  } as const);

export const SET_BASE_URL = 'SET_BASE_URL';
export const setBaseUrl = (baseUrl: State['baseUrl']) =>
  ({
    type: SET_BASE_URL,
    payload: {
      baseUrl
    }
  } as const);

export const RESET_PAGES_STATUS = 'RESET_PAGES_STATUS';
export const resetPageStatus = () =>
  ({
    type: RESET_PAGES_STATUS
  } as const);

export const LOAD_APP_PROCESS = 'LOAD_APP_PROCESS';
export const loadAppProcess = () =>
  ({
    type: LOAD_APP_PROCESS
  } as const);

export const LOAD_APP_PROCESS_SUCCESS = 'LOAD_APP_PROCESS_SUCCESS';
export const loadAppProcessSuccess = () =>
  ({
    type: LOAD_APP_PROCESS_SUCCESS
  } as const);

export const LOAD_APP_PROCESS_FAILURE = 'LOAD_APP_PROCESS_FAILURE';
export const loadAppProcessFailure = (err: Error) =>
  ({
    type: LOAD_APP_PROCESS_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_HOME_PAGE = 'LOAD_HOME_PAGE';
export const loadHomePage = () =>
  ({
    type: LOAD_HOME_PAGE
  } as const);

export const LOAD_HOME_PAGE_SUCCESS = 'LOAD_HOME_PAGE_SUCCESS';
export const loadHomePageSuccess = () =>
  ({
    type: LOAD_HOME_PAGE_SUCCESS
  } as const);

export const LOAD_HOME_PAGE_FAILURE = 'LOAD_HOME_PAGE_FAILURE';
export const loadHomePageFailure = (err: Error) =>
  ({
    type: LOAD_HOME_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);

export const LOAD_DOCUMENTATION_PAGE = 'LOAD_DOCUMENTATION_PAGE';
export const loadDocumentationPage = () =>
  ({
    type: LOAD_DOCUMENTATION_PAGE
  } as const);

export const LOAD_DOCUMENTATION_PAGE_SUCCESS = 'LOAD_DOCUMENTATION_PAGE_SUCCESS';
export const loadDocumentationPageSuccess = () =>
  ({
    type: LOAD_DOCUMENTATION_PAGE_SUCCESS
  } as const);

export const LOAD_DOCUMENTATION_PAGE_FAILURE = 'LOAD_DOCUMENTATION_PAGE_FAILURE';
export const loadDocumentationPageFailure = (err: Error) =>
  ({
    type: LOAD_DOCUMENTATION_PAGE_FAILURE,
    payload: {
      err
    }
  } as const);
