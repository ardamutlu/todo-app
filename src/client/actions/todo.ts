export type Actions =
  | ReturnType<typeof getAllTodoRequest>
  | ReturnType<typeof getAllTodoSuccess>
  | ReturnType<typeof getAllTodoFailure>
  | ReturnType<typeof deleteTodoByIdRequest>
  | ReturnType<typeof deleteTodoByIdSuccess>
  | ReturnType<typeof deleteTodoByIdFailure>
  | ReturnType<typeof updateTodoStatusByIdRequest>
  | ReturnType<typeof updateTodoStatusByIdSuccess>
  | ReturnType<typeof updateTodoStatusByIdFailure>
  | ReturnType<typeof addNewTodoRequest>
  | ReturnType<typeof addNewTodoSuccess>
  | ReturnType<typeof addNewTodoFailure>
  | ReturnType<typeof updateTodoByIdRequest>
  | ReturnType<typeof updateTodoByIdSuccess>
  | ReturnType<typeof updateTodoByIdFailure>
  | ReturnType<typeof filterTodo>
  | ReturnType<typeof searchTodo>;

/**
 * GET ALL TODO
 */
export const GET_ALL_TODO_REQUEST = 'GET_ALL_TODO_REQUEST';
export const getAllTodoRequest = () =>
  ({
    type: GET_ALL_TODO_REQUEST
  } as const);

export const GET_ALL_TODO_SUCCESS = 'GET_ALL_TODO_SUCCESS';
export const getAllTodoSuccess = (payload: any) =>
  ({
    type: GET_ALL_TODO_SUCCESS,
    payload
  } as const);

export const GET_ALL_TODO_FAILURE = 'GET_ALL_TODO_FAILURE';
export const getAllTodoFailure = (err: Error) =>
  ({
    type: GET_ALL_TODO_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * DELETE TODO BY ID
 */
export const DELETE_TODO_BY_ID_REQUEST = 'DELETE_TODO_BY_ID_REQUEST';
export const deleteTodoByIdRequest = (payload: number) =>
  ({
    type: DELETE_TODO_BY_ID_REQUEST,
    payload
  } as const);

export const DELETE_TODO_BY_ID_SUCCESS = 'DELETE_TODO_BY_ID_SUCCESS';
export const deleteTodoByIdSuccess = (payload: any) =>
  ({
    type: DELETE_TODO_BY_ID_SUCCESS,
    payload
  } as const);

export const DELETE_TODO_BY_ID_FAILURE = 'DELETE_TODO_BY_ID_FAILURE';
export const deleteTodoByIdFailure = (err: Error) =>
  ({
    type: DELETE_TODO_BY_ID_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * UPDATE TODO STATUS BY ID
 */
export const UPDATE_TODO_STATUS_BY_ID_REQUEST = 'UPDATE_TODO_STATUS_BY_ID_REQUEST';
export const updateTodoStatusByIdRequest = (payload: any) =>
  ({
    type: UPDATE_TODO_STATUS_BY_ID_REQUEST,
    payload
  } as const);

export const UPDATE_TODO_STATUS_BY_ID_SUCCESS = 'UPDATE_TODO_STATUS_BY_ID_SUCCESS';
export const updateTodoStatusByIdSuccess = (payload: any) =>
  ({
    type: UPDATE_TODO_STATUS_BY_ID_SUCCESS,
    payload
  } as const);

export const UPDATE_TODO_STATUS_BY_ID_FAILURE = 'UPDATE_TODO_STATUS_BY_ID_FAILURE';
export const updateTodoStatusByIdFailure = (err: Error) =>
  ({
    type: UPDATE_TODO_STATUS_BY_ID_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * ADD NEW TODO
 */
export const ADD_NEW_TODO_REQUEST = 'ADD_NEW_TODO_REQUEST';
export const addNewTodoRequest = (payload: any) =>
  ({
    type: ADD_NEW_TODO_REQUEST,
    payload
  } as const);

export const ADD_NEW_TODO_SUCCESS = 'ADD_NEW_TODO_SUCCESS';
export const addNewTodoSuccess = (payload: any) =>
  ({
    type: ADD_NEW_TODO_SUCCESS,
    payload
  } as const);

export const ADD_NEW_TODO_FAILURE = 'ADD_NEW_TODO_FAILURE';
export const addNewTodoFailure = (err: Error) =>
  ({
    type: ADD_NEW_TODO_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * UPDATE TODO BY ID
 */
export const UPDATE_TODO_BY_ID_REQUEST = 'UPDATE_TODO_BY_ID_REQUEST';
export const updateTodoByIdRequest = (payload: any) =>
  ({
    type: UPDATE_TODO_BY_ID_REQUEST,
    payload
  } as const);

export const UPDATE_TODO_BY_ID_SUCCESS = 'UPDATE_TODO_BY_ID_SUCCESS';
export const updateTodoByIdSuccess = (payload: any) =>
  ({
    type: UPDATE_TODO_BY_ID_SUCCESS,
    payload
  } as const);

export const UPDATE_TODO_BY_ID_FAILURE = 'UPDATE_TODO_BY_ID_FAILURE';
export const updateTodoByIdFailure = (err: Error) =>
  ({
    type: UPDATE_TODO_BY_ID_FAILURE,
    payload: {
      err
    }
  } as const);

/**
 * FILTER TODO
 */
export const FILTER_TODO = 'FILTER_TODO';
export const filterTodo = (payload: any) =>
  ({
    type: FILTER_TODO,
    payload
  } as const);

/**
 * SEARCH TODO
 */
export const SEARCH_TODO = 'SEARCH_TODO';
export const searchTodo = (query: string) =>
  ({
    type: SEARCH_TODO,
    payload: {
      query
    }
  } as const);
