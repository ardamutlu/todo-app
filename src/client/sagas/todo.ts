import { END } from 'redux-saga';
import { put, call, takeLatest } from 'redux-saga/effects';
import {
  getAllTodoRequest,
  getAllTodoSuccess,
  getAllTodoFailure,
  deleteTodoByIdRequest,
  deleteTodoByIdSuccess,
  deleteTodoByIdFailure,
  updateTodoStatusByIdRequest,
  updateTodoStatusByIdSuccess,
  updateTodoStatusByIdFailure,
  updateTodoByIdRequest,
  updateTodoByIdSuccess,
  updateTodoByIdFailure,
  addNewTodoRequest,
  addNewTodoSuccess,
  addNewTodoFailure,
  GET_ALL_TODO_REQUEST,
  DELETE_TODO_BY_ID_REQUEST,
  UPDATE_TODO_STATUS_BY_ID_REQUEST,
  UPDATE_TODO_BY_ID_REQUEST,
  ADD_NEW_TODO_REQUEST
} from '../actions/todo';
import { fetchMethod } from './fetchMethod';
import { ErrorNotification } from '../templates';

function* getAllTodo(actions: ReturnType<typeof getAllTodoRequest>) {
  const options = { method: 'GET' };

  try {
    // async

    const body = yield fetchMethod({
      path: `/`,
      options
    });

    yield put(getAllTodoSuccess(body));
  } catch (err) {
    yield put(getAllTodoFailure(err));
  } finally {
    if (!process.env.IS_BROWSER) {
      yield call(stopSaga);
    }
  }
}

function* deleteTodoById(actions: ReturnType<typeof deleteTodoByIdRequest>) {
  const options = { method: 'DELETE' };

  try {
    // async

    const body = yield fetchMethod({
      path: `/${actions.payload}`,
      options
    });

    if (body.error) {
      yield call(ErrorNotification, 'error', body.message);
      yield put(deleteTodoByIdFailure(body));

      return;
    }

    yield put(deleteTodoByIdSuccess(body));
    yield call(ErrorNotification, 'success', 'Todo deleted successfully.');
  } catch (err) {
    yield put(deleteTodoByIdFailure(err));
  } finally {
    if (!process.env.IS_BROWSER) {
      yield call(stopSaga);
    }
  }
}

function* updateTodoStatusById(actions: ReturnType<typeof updateTodoStatusByIdRequest>) {
  const options = {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(
      Object.assign({}, actions.payload, { complete: !actions.payload.complete })
    )
  };

  try {
    // async

    const body = yield fetchMethod({
      path: `/${actions.payload.id}`,
      options
    });

    if (body.error) {
      yield call(ErrorNotification, 'error', body.message);
      yield put(updateTodoStatusByIdFailure(body));

      return;
    }

    yield put(updateTodoStatusByIdSuccess(body));
    yield call(ErrorNotification, 'success', 'Status changed successfully.');
  } catch (err) {
    yield put(updateTodoStatusByIdFailure(err));
  } finally {
    if (!process.env.IS_BROWSER) {
      yield call(stopSaga);
    }
  }
}

function* updateTodoById(actions: ReturnType<typeof updateTodoByIdRequest>) {
  const options = {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(actions.payload)
  };

  try {
    // async

    const body = yield fetchMethod({
      path: `/${actions.payload.id}`,
      options
    });

    if (body.error) {
      yield call(ErrorNotification, 'error', body.message);
      yield put(updateTodoByIdFailure(body));
      return;
    }

    yield put(updateTodoByIdSuccess(body));
    yield call(ErrorNotification, 'success', 'Todo updated successfully.');
  } catch (err) {
    yield put(updateTodoByIdFailure(err));
  } finally {
    if (!process.env.IS_BROWSER) {
      yield call(stopSaga);
    }
  }
}

function* addNewTodo(actions: ReturnType<typeof addNewTodoRequest>) {
  const options = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(actions.payload)
  };

  try {
    // async

    const body = yield fetchMethod({
      path: `/`,
      options
    });

    if (body.error) {
      yield call(ErrorNotification, 'error', body.message);
      yield put(addNewTodoFailure(body));
      return;
    }

    yield put(addNewTodoSuccess(body));
    yield call(ErrorNotification, 'success', 'Todo created successfully.');
  } catch (err) {
    yield put(addNewTodoFailure(err));
  } finally {
    if (!process.env.IS_BROWSER) {
      yield call(stopSaga);
    }
  }
}

function* stopSaga() {
  yield put(END);
}

export function* todoProcess() {
  yield takeLatest(GET_ALL_TODO_REQUEST, getAllTodo);
  yield takeLatest(DELETE_TODO_BY_ID_REQUEST, deleteTodoById);
  yield takeLatest(UPDATE_TODO_STATUS_BY_ID_REQUEST, updateTodoStatusById);
  yield takeLatest(UPDATE_TODO_BY_ID_REQUEST, updateTodoById);
  yield takeLatest(ADD_NEW_TODO_REQUEST, addNewTodo);
}
