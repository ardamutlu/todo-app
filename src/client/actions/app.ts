import { State } from '../reducers/pages';

export type Actions = ReturnType<typeof setIsMobile>;

export const SET_IS_MOBILE = 'SET_IS_MOBILE';
export const setIsMobile = (isMobile: State['isMobile']) =>
  ({
    type: SET_IS_MOBILE,
    payload: {
      isMobile
    }
  } as const);
