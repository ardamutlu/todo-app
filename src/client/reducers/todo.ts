import * as actions from '../actions/todo';

export type State = {
  entities: Array<{}>;
  query: string;
  filter: {
    value: number;
    title: string;
  };
  loading: boolean;
  error: Error | null;
};

export const initialState: State = {
  entities: [],
  query: '',
  filter: {
    value: 0,
    title: 'All'
  },
  loading: false,
  error: null
};

export const reducer = (state: State = initialState, action: actions.Actions): State => {
  switch (action.type) {
    case actions.GET_ALL_TODO_REQUEST:
    case actions.DELETE_TODO_BY_ID_REQUEST:
    case actions.UPDATE_TODO_STATUS_BY_ID_REQUEST:
    case actions.UPDATE_TODO_BY_ID_REQUEST:
    case actions.ADD_NEW_TODO_REQUEST:
      return { ...state, loading: true };

    case actions.GET_ALL_TODO_SUCCESS:
    case actions.DELETE_TODO_BY_ID_SUCCESS:
    case actions.UPDATE_TODO_STATUS_BY_ID_SUCCESS:
    case actions.UPDATE_TODO_BY_ID_SUCCESS:
    case actions.ADD_NEW_TODO_SUCCESS:
      return { ...state, loading: false, entities: action.payload };

    case actions.GET_ALL_TODO_FAILURE:
    case actions.DELETE_TODO_BY_ID_FAILURE:
    case actions.UPDATE_TODO_STATUS_BY_ID_FAILURE:
    case actions.UPDATE_TODO_BY_ID_FAILURE:
    case actions.ADD_NEW_TODO_FAILURE:
      return { ...state, loading: false, error: action.payload.err };

    case actions.FILTER_TODO:
      return { ...state, filter: action.payload };

    case actions.SEARCH_TODO:
      return { ...state, query: action.payload.query };

    default:
      return state;
  }
};
