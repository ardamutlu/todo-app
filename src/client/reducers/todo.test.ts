import { reducer, initialState } from './todo';
import * as actions from '../actions/todo';

test('should return the initial state', () => {
  expect(reducer(undefined, { type: undefined } as any)).toEqual(initialState);
});

test('should handle GET_ALL_TODO_REQUEST', () => {
  expect(reducer(initialState, actions.getAllTodoRequest())).toMatchSnapshot();
});
test('should handle GET_ALL_TODO_SUCCESS', () => {
  expect(reducer(initialState, actions.getAllTodoSuccess([]))).toMatchSnapshot();
});
test('should handle GET_ALL_TODO_FAILURE', () => {
  expect(reducer(initialState, actions.getAllTodoFailure(new Error('404')))).toMatchSnapshot();
});

test('should handle DELETE_TODO_BY_ID_REQUEST', () => {
  expect(reducer(initialState, actions.deleteTodoByIdRequest(0))).toMatchSnapshot();
});
test('should handle DELETE_TODO_BY_ID_SUCCESS', () => {
  expect(reducer(initialState, actions.deleteTodoByIdSuccess([]))).toMatchSnapshot();
});
test('should handle DELETE_TODO_BY_ID_FAILURE', () => {
  expect(reducer(initialState, actions.deleteTodoByIdFailure(new Error('404')))).toMatchSnapshot();
});

test('should handle UPDATE_TODO_STATUS_BY_ID_REQUEST', () => {
  expect(reducer(initialState, actions.updateTodoStatusByIdRequest({}))).toMatchSnapshot();
});
test('should handle UPDATE_TODO_STATUS_BY_ID_SUCCESS', () => {
  expect(reducer(initialState, actions.updateTodoStatusByIdSuccess([]))).toMatchSnapshot();
});
test('should handle UPDATE_TODO_STATUS_BY_ID_FAILURE', () => {
  expect(
    reducer(initialState, actions.updateTodoStatusByIdFailure(new Error('404')))
  ).toMatchSnapshot();
});

test('should handle ADD_NEW_TODO_REQUEST', () => {
  expect(reducer(initialState, actions.addNewTodoRequest({}))).toMatchSnapshot();
});
test('should handle ADD_NEW_TODO_SUCCESS', () => {
  expect(reducer(initialState, actions.addNewTodoSuccess([]))).toMatchSnapshot();
});
test('should handle ADD_NEW_TODO_FAILURE', () => {
  expect(reducer(initialState, actions.addNewTodoFailure(new Error('404')))).toMatchSnapshot();
});

test('should handle UPDATE_TODO_BY_ID_REQUEST', () => {
  expect(reducer(initialState, actions.updateTodoByIdRequest({}))).toMatchSnapshot();
});
test('should handle UPDATE_TODO_BY_ID_SUCCESS', () => {
  expect(reducer(initialState, actions.updateTodoByIdSuccess([]))).toMatchSnapshot();
});
test('should handle UPDATE_TODO_BY_ID_FAILURE', () => {
  expect(reducer(initialState, actions.updateTodoByIdFailure(new Error('404')))).toMatchSnapshot();
});

test('should handle FILTER_TODO', () => {
  expect(reducer(initialState, actions.filterTodo({}))).toMatchSnapshot();
});

test('should handle SEARCH_TODO', () => {
  expect(reducer(initialState, actions.searchTodo(''))).toMatchSnapshot();
});
