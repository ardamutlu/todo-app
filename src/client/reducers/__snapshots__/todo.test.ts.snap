// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle ADD_NEW_TODO_FAILURE 1`] = `
Object {
  "entities": Array [],
  "error": [Error: 404],
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle ADD_NEW_TODO_REQUEST 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": true,
  "query": "",
}
`;

exports[`should handle ADD_NEW_TODO_SUCCESS 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle DELETE_TODO_BY_ID_FAILURE 1`] = `
Object {
  "entities": Array [],
  "error": [Error: 404],
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle DELETE_TODO_BY_ID_REQUEST 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": true,
  "query": "",
}
`;

exports[`should handle DELETE_TODO_BY_ID_SUCCESS 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle FILTER_TODO 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {},
  "loading": false,
  "query": "",
}
`;

exports[`should handle GET_ALL_TODO_FAILURE 1`] = `
Object {
  "entities": Array [],
  "error": [Error: 404],
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle GET_ALL_TODO_REQUEST 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": true,
  "query": "",
}
`;

exports[`should handle GET_ALL_TODO_SUCCESS 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle SEARCH_TODO 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle UPDATE_TODO_BY_ID_FAILURE 1`] = `
Object {
  "entities": Array [],
  "error": [Error: 404],
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle UPDATE_TODO_BY_ID_REQUEST 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": true,
  "query": "",
}
`;

exports[`should handle UPDATE_TODO_BY_ID_SUCCESS 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle UPDATE_TODO_STATUS_BY_ID_FAILURE 1`] = `
Object {
  "entities": Array [],
  "error": [Error: 404],
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;

exports[`should handle UPDATE_TODO_STATUS_BY_ID_REQUEST 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": true,
  "query": "",
}
`;

exports[`should handle UPDATE_TODO_STATUS_BY_ID_SUCCESS 1`] = `
Object {
  "entities": Array [],
  "error": null,
  "filter": Object {
    "title": "All",
    "value": 0,
  },
  "loading": false,
  "query": "",
}
`;
